src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -320,24 +338,28 @@ uiModules.get('apps/management')
 
     let lastPromise;
     resetIndex();
-    samplePromise = lastPromise = updateSamples()
+    samplePromise = lastPromise = Promise.resolve()
```

Comment:
This whole `samplePromise` and `lastPromise` is very confusing. While you're in here, do you mind adding some clarifying comments explaining what these do and why they are necessary?